frontend/src/containers/Billings/BillingForm__SecondStep.tsx

```diff
@@ -0,0 +1,205 @@
+    (member) => member.id !== billingForm.charged_member_id
+  );
+
+  const defaultCheckedMembers = filteredMembers.map((_) => false);
+
+  const [checkedMembers, setCheckedMembers] = React.useState(
+    defaultCheckedMembers
+  );
+
+  // React.useEffect(() => {
+  //   if (formMode === FormMode.Create) {
+  // const selectedMembers = filteredMembers.map((member, index) => {
+  //   const selectedMember = checkedMembers[index] ? member.id : "";
+  //   return {
+  //     id: selectedMember,
+  //     amount: 0,
+  //   };
+  // });
+  //     send({
+  //       type: "UPDATE_FORM",
+  //       billingForm: {
+  //         ...billingForm,
+  //         members: selectedMembers,
+  //       },
+  //     });
+  //   }
+  // }, [billingForm, checkedMembers, filteredMembers, formMode, send]);
+
+  return (
+    <Modal isOpen={true} onClose={handleCloseForm} isCentered>
+      <ModalOverlay
+        bg="blackAlpha.300"
+        backdropFilter="blur(10px) hue-rotate(90deg)"
+      />
+      <ModalContent>
+        <ModalHeader textAlign="center">Billing Form</ModalHeader>
+        <ModalCloseButton />
+        <ModalBody pb={6}>
+          <Flex justifyContent="space-between">
+            <Flex>
+              <Text size="md" mr="2">
+                Total Bill
+              </Text>
+              <Text fontWeight="bold">Rp. {billingForm.bill_amount}</Text>
+            </Flex>
+            <Checkbox
+              size="lg"
+              colorScheme="cyan"
+              isChecked={billingForm?.is_bill_equally}
+              onChange={(event) =>
+                send({
+                  type: "UPDATE_FORM",
+                  billingForm: {
+                    ...billingForm,
+                    is_bill_equally: event.target.checked,
+                  },
+                })
+              }
+            >
+              Bill Equally
+            </Checkbox>
+          </Flex>
+          <VStack spacing="4" my="4">
+            {filteredMembers.map((member, index) => (
+              <Box
+                key={member.id}
+                border="1px"
+                borderRadius="16px"
+                p="6"
+                w="full"
+                backgroundColor="gray.700"
+              >
+                <Flex justifyContent="space-between">
+                  <Box>
+                    <Text>{member.name}</Text>
+                    {!billingForm.is_bill_equally && checkedMembers[index] && (
+                      <Input
+                        type="number"
+                        placeholder="Input the amount of bill"
+                        borderColor="whiteAlpha.600"
+                        mt="2"
+                        value={billingForm.members[index]?.amount}
+                        onChange={(event) => {
+                          billingForm.members[index].amount =
+                            event.target.valueAsNumber;
+                          send({ type: "UPDATE_FORM", billingForm });
```

Comment:
ini object `billingForm` jangan dimutate ya, dan jangan berdasarkan index ngubahnya, jadi bisa kayak gini aja : 

```
const updatedBillingForm = {
  ...billingForm,
  members: {
    [member.id]: event.target.valueAsNumber
  }
}
send({ type: "UPDATE_FORM", updatedBillingForm  });
```

Reply:
Solved in huddle discussion, since the members are an array, we can implement this update mechanism =>
```
                            const updatedBillingForm = {
                              ...billingForm,
                              members: billingForm.members.map(
                                ({ id, amount }) => ({
                                  id,
                                  amount:
                                    member.id === id
                                      ? event.target.valueAsNumber
                                      : amount,
                                })
                              ),
                            };
                            send({
                              type: "UPDATE_FORM",
                              billingForm: updatedBillingForm,
                            });
```